lib/instrumentation/modules/memcached.js

```diff
@@ -0,0 +1,49 @@
+'use strict'
+
+var semver = require('semver')
+
+var shimmer = require('../shimmer')
+
+module.exports = function (memcached, agent, { version, enabled }) {
+  if (!enabled) {
+    return memcached
+  }
+  if (!semver.satisfies(version, '>=2.2.0')) {
+    agent.logger.debug('Memcached version %s not supported - aborting...', version)
+    return memcached
+  }
+  agent.logger.debug('shimming memcached.prototype.command')
+  shimmer.wrap(memcached.prototype, 'command', wrapCommand)
+  return memcached
+
+  // Wrap the generic command that is used to build touch, get, gets etc
+  function wrapCommand (original) {
+    return function wrappedCommand () {
+      if (typeof arguments[0] === 'function') {
+        var query = arguments[0]()
+        // If the callback is not a function the user doesn't care about result
+        if (query && typeof query.callback === 'function') {
+          var span = agent.startSpan(`memcached.${query.type}`, `db.memcached.${query.type}`)
+          if (span) {
+            span.setDbContext({ statement: `${query.type} ${query.key}`, type: 'memcached' })
+            var id = span.id
```

Comment:
Not a fan of using a nested `var` that escapes its scope like this. Could we maybe update this file to use `const`/`let` instead and avoid this weird pattern?

Reply:
I'm happy to use let and move the section that uses span.id inside the scope so its not escaped. The other modules I've looked at briefly for inspiration are using var though so I thought it was a standard.

Reply:
This code is technically fine but could be easier to read if we moved the variable declaration outside of the if-statement. I didn't comment on the use of `var` as I know we have some very old code from the Node.js 0.8 days still around in our code-base. And I knew you probably got inspired from some of that 😅 

Today I would define `id` outside of the if-statement using `let` and set it just like today inside the if-statement.